src/Tribe/Commerce/PayPal/Gateway.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+class Tribe__Tickets__Commerce__PayPal__Gateway {
+	/**
+	 * @var string
+	 */
+	public $identity_token;
+
+	/**
+	 * @var array
+	 */
+	protected $transaction_data;
+
+	/**
+	 * Tribe__Tickets__Commerce__PayPal__Gateway constructor.
+	 *
+	 * @since TBD
+	 */
+	public function __construct() {
+		$this->hook();
+		$this->identity_token = tribe_get_option( 'ticket-paypal-identity-token' );
+
+		if ( $this->identity_token ) {
+			// if there's an identity token set, we handle payment confirmation with PDT
+			tribe( 'tickets.commerce.paypal.handler.pdt' );
+		} else {
+			// if there isn't an identity token set, we use IPN
+			tribe( 'tickets.commerce.paypal.handler.ipn' );
+		}
+	}
+
+	/**
+	 * Set up hooks for the gateway
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		add_action( 'template_redirect', array( $this, 'add_to_cart' ) );
+	}
+
+	/**
+	 * Handles adding tickets to cart
+	 *
+	 * @since TBD
+	 */
+	public function add_to_cart() {
+		global $post;
+
+		// bail if this isn't a Tribe Commerce PayPal ticket
+		if (
+			empty( $_POST['product_id'] )
+			|| empty( $_POST['provider'] )
+			|| 'Tribe__Tickets__Commerce__PayPal__Main' !== $_POST['provider']
+		) {
+			return;
+		}
+
+		$url           = $this->get_cart_url( '_cart' );
+		$now           = time();
+		$post_url      = get_permalink( $post );
+		$currency_code = tribe_get_option( 'ticket-paypal-currency-code' );
+
+		// @TODO: set to the success page
```

Comment:
okay for this `@todo` to remain here at this point?

Reply:
Yeah, this feature isn't done yet so that's there for future work.